packages/flutter_tools/lib/src/commands/start.dart

```diff
@@ -109,10 +117,12 @@ class StartCommand extends StartCommandBase {
 }
 
 Future<int> startApp(
+  BuildConfiguration config,
```

Comment:
Can you move this to after `toolchain`? Where we need these args, we generally use `devices`, `applicationPackages`, and `toolchain` in that order.

(though we should look into ways to not have to pass around quite so many variables)